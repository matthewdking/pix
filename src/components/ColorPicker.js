import React, { Component } from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { fetchPictures } from '../actions/fetchPictures';

const ColorContainer = styled.section`
  display: flex;
  justify-content: space-around;
`;

const SingleColorWrapper = styled.div`
  height: 3rem;
  width: 3rem;
  display: inline-block;
  cursor: pointer;
`;

const SingleColor = styled.div`
  background-color: ${props => props.color};
  height: 5rem;
  width: 5rem;
  text-align: center;
  line-height: 5rem;
  font-size: 1.4rem;
  border-radius: 1rem;
  border: 0.15rem black solid;
`;

export const randomSearch = searchQueries => {
  return searchQueries[Math.floor(Math.random() * searchQueries.length)];
};

export class ColorPicker extends Component {
  constructor(props) {
    super(props);

    this.state = {
      colors: [
        {
          name: 'Red',
          hex: '#F44336',
        },
        {
          name: 'Yellow',
          hex: '#FFEB3B',
        },
        {
          name: 'Green',
          hex: '#4CAF50',
        },
        {
          name: 'Purple',
          hex: '#9C27B0',
        },
      ],
      searchQueries: [
        'man',
        'mountain',
        'state',
        'ocean',
        'country',
        'building',
        'cat',
        'airline',
        'wealth',
        'happiness',
        'pride',
        'fear',
        'religion',
        'bird',
        'book',
        'phone',
        'rice',
        'snow',
        'water',
      ],
    };
  }

  render() {
    return (
      <ColorContainer>
        {this.state.colors.map((color, index) => (
          <SingleColorWrapper
            key={index}
            onClick={() => {
              this.props.fetchPictures(
                color.name,
                randomSearch(this.state.searchQueries),
              );
            }}
          >
            <SingleColor color={color.hex}>{color.name}</SingleColor>
          </SingleColorWrapper>
        ))}
      </ColorContainer>
    );
  }
}

const mapDispatchToProps = {
  fetchPictures,
};

const mapStateToProps = state => ({
  pictures: state.pictures,
});

export default connect(mapStateToProps, mapDispatchToProps)(ColorPicker);
