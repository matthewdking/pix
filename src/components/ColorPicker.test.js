import React from 'react';
import renderer from 'react-test-renderer';
import { ColorPicker, randomSearch } from './ColorPicker';

Math.random = jest.fn(() => 0.5);
// Math.random produces 0.5 for all tests

const searchQueries = [
  'man',
  'mountain',
  'state',
  'ocean',
  'country',
  'building',
  'cat',
  'airline',
  'wealth',
  'happiness',
  'pride',
  'fear',
  'religion',
  'bird',
  'book',
  'phone',
  'rice',
  'snow',
  'water',
];

const store = test('randomSearch returns middle value', () => {
  expect(randomSearch([1, 2, 3])).toBe(2);
  expect(randomSearch(['hello'])).toBe('hello');
  expect(randomSearch(searchQueries)).toBe('happiness');
});

test('ColorPicker snapshot', () => {
  const component = renderer.create(<ColorPicker />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
