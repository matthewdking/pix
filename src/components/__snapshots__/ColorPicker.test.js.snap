// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker snapshot 1`] = `
<section
  className="sc-bdVaJa imDAUP"
>
  <div
    className="sc-bwzfXH bVOPkM"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat MxFAq"
      color="#F44336"
    >
      Red
    </div>
  </div>
  <div
    className="sc-bwzfXH bVOPkM"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat jOaskf"
      color="#FFEB3B"
    >
      Yellow
    </div>
  </div>
  <div
    className="sc-bwzfXH bVOPkM"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat jpwNYZ"
      color="#4CAF50"
    >
      Green
    </div>
  </div>
  <div
    className="sc-bwzfXH bVOPkM"
    onClick={[Function]}
  >
    <div
      className="sc-htpNat cOTfMF"
      color="#9C27B0"
    >
      Purple
    </div>
  </div>
</section>
`;
